src/ucs/async/signal.c

```diff
@@ -181,6 +181,29 @@ static ucs_status_t ucs_async_signal_dispatch_timer(int uid)
     return ucs_async_dispatch_timerq(&timer->timerq, ucs_get_time());
 }
 
+static inline int ucs_signal_map_to_events(int si_code)
+{
+    int events = 0;
+
+    switch (si_code) {
+    case POLL_IN:
+    case POLL_MSG:
+    case POLL_PRI:
+        events |= UCS_EVENT_SET_EVREAD;
+        return events;
+    case POLL_OUT:
+        events |= UCS_EVENT_SET_EVWRITE;
+        return events;
+    case POLL_HUP:
+    case POLL_ERR:
+        events |= UCS_EVENT_SET_EVERR;
+        return events;
+    default:
+        ucs_warn("unexpected si_code %d", si_code);
```

Comment:
so it would be printed twice, also in ucs_async_signal_handler?

Reply:
@yosefe no. in ucs_async_signal_handler it is called before the 'default' case so in this case, no warning would be printed at all.
i think that ucs_signal_map_to_events should be a standalone function and print a warning on an unknown si_code.

Reply:
ok